Classes_HW4.py

```diff
@@ -0,0 +1,208 @@
+    print('{0:25s}'.format("-----------------------------------------------------------------------------------------" ))
+    for curPhase in phaseList:
+        print('{0:20s}{1:11.1f}{2:11.1f}{3:11.1f}{4:13.1f}{5:13.1f}{6:14.1f}'.format(curPhase.strName, curPhase.dvPhase, curPhase.mStart, curPhase.mEnd, curPhase.mPropImpulse, curPhase.mPropImpulseOx, curPhase.mPropImpulseFuel ))
+
+class Phase:
+
+    def __init__(self,strName, mStart, dvPhase, clsEng):
+
+        # Check if this is a T/W phase. If so, 
+        # update the dV calculate the thrust-to-weight
+        twPhase = clsEng.thrust/(mStart*9.81)
+        if dvPhase<0:
+            dvPhase = 4335*np.exp(-(twPhase)*20.25)+1880
+
+
+        # Calculate Impulse Propellant Using Rocket Equation   
+        mPropImpulse = mStart - mStart / np.exp(dvPhase/(9.81*clsEng.isp))
+
+   
+        # Determine Oxidizer and Fuel
+        mPropImpulseOx = (mPropImpulse * clsEng.mr) / (1 + clsEng.mr)
+        mPropImpulseFuel = mPropImpulse / (1 + clsEng.mr)
+        
+          
+        mEnd = mStart - mPropImpulse 
+        
+        
+        # Move data to class structure to save information
+        self.mStart         = mStart
+        self.mEnd           = mEnd
+        self.dvPhase        = dvPhase
+        self.clsEng         = clsEng
+        self.mPropImpulse   = mPropImpulse
+        self.strName        = strName
+        self.twPhase        = twPhase
+
+        self.mPropImpulse   = mPropImpulse
+        self.mPropImpulseOx = mPropImpulseOx
+        self.mPropImpulseFuel = mPropImpulseFuel
+          
+class MissionSummary:
+    def __init__(self, tupPhases):
+        """
+        Inputs:
+            tupPhases: list of phase classes
+        
+        """
+
+   
+        
+        mPropImpulse     = 0
+        mPropImpulseOx   = 0
+        mPropImpulseFuel = 0
+
+        # sum up the usages by phase
+        for curPhase in tupPhases:
+            mPropImpulse     += curPhase.mPropImpulse 
+            mPropImpulseOx   += curPhase.mPropImpulseOx
+            mPropImpulseFuel += curPhase.mPropImpulseFuel
+
+        # Stuff everything into self    
+        self.mPropImpulse      = mPropImpulse
+        self.mPropImpulseOx    = mPropImpulseOx
+        self.mPropImpulseFuel  = mPropImpulseFuel
+
+
+
+class Engine:
+    def __init__(self,isp, thrust, mr):
+        self.isp = isp
+        self.thrust = thrust
+        self.mr = mr
+
+class TankSet:
+    def __init__(self, strPropType,strMatType, nTanks, lMaxRadTank, presTank, mPropTotal):
+        # General Parameters for Tanks
+        pctUllage =   1.1  # Extra ullage room as a percentage of tank volume
+        aMax      =   50      # Maximum acceleration (m/s2)
+        pctFudge  =   1.2      # Fudge factor for welds, etc
+        fosMat    =   1.4      # factor of safety for material (nd)

```

Comment:
Your factor of safety should be 1.5, not 1.4. When I changed that you passed the unit tests